127 Add entry field for Associated Datasets

fostermh opened the pull request:
https://github.com/cioos-siooc/metadata-entry-form/issues/127

first try at adding a 'related' tab to list related datasets. There are some issues here with trying to accommodate all datacite identifiers as many of them are not necessarily urls. I'm not sure the best way to go here. If we should stick to the iso standard and allow anything in the identfyer type field or restrict to the list datacite provides?

fostermh commented:
All changes other then merging the tabs have been completed

see https://github.com/cioos-siooc/metadata-entry-form/issues/127 for a discussion on why we may or may not want to merge resources and related tabs.

Br-Johnson reviewed (changes requested):
- Related identifier field is required to be a full URL, but ISNN for example is not a URL. Remove requirement for it to be a full URL?
- Change 'Related' tab to 'Related Works'
- Word smith 'Resource' tab docs to identify that 'Resources' are the primary 'resource' that the metadata record describes.
- Word smith 'Related Works' documentation to make it clear this is for works published elsewhere

fostermh commented:
I think I have captured all the changes requested

fostermh commented:
ok, @timvdstap I think I have addressed your comments.

- translation fixed, the function was deployed without necessary aws credentials, my bad
- example of using 'Dependency' added
- yes, supporting documents such as sampling protocols would still go under resources. other cited works such as a published paper, a service built using data from this record, or an aggregated dataset of which this is a part would go under related works.
- validation now works. I did not know about the submit page or had forgotten, thanks for checking that.

Br-Johnson reviewed (changes requested):
Some rewording to consistently use 'primary resource' to identify the resource the current metadata record being created is referring to, and to use related work to identify the 'associated dataset/resource' people are linking to. Also I tried to replace 'records' with either one of the previously mentioned terms.

Tried clarifying the language for relation Type to make it as clear as possible.

I also realized maybe you could include resource type (ie [MD_ScopeCode](https://wiki.esipfed.org/ISO_19115-3_Codelists#MD_ScopeCode)) for the primary resource. Maybe thats another issue though if it's not simple to implement.

fostermh commented:
I'm not sure that mapping 'crossReference' directly to cited is a good idea. In my mind crossReference is also for datasets that are siblings and part of the same collection but not necessarily citing each other. 

I found this [crosswalk](https://drive.google.com/open?id=1P6WH8h4OnIVR9UJj3FcOebNUpLnKNBCuvEp3NsLRho4) which has some suggestions on this site https://www.rd-alliance.org/groups/research-metadata-schemas-wg if you search for 'associationType' a few different things come up.

One mapping shows 'isReferencedBy' mapping to associated resources but suggests using the datacite list of relation types. One option would be to add a specific type to our list for example 'Cites' to populate datacite metadata with. When generating the iso19115 xml we would have to map 'Cites' back to something like crossReference. This is the only way I can see to capture all the info you are asking for while still staying true to the ISO standard.

fostermh commented:
I also wonder if mdb:MD_Metadata/mdb:resourceLineage/mrl:LI_Lineage/mrl:source/mrl:LI_Source/mrl:sourceCitation/cit:CI_Citation would be a better place to put things that are cited by this resource and thus contributed to its creation (lineage)
https://wiki.esipfed.org/LI_Source

fostermh commented:
TODO: change relationship type to use the datacite types and map these back to the iso standard.

Br-Johnson commented:
> TODO: change relationship type to use the datacite types and map these back to the iso standard.

Here's a google sheet where we can map the terms: https://docs.google.com/spreadsheets/d/1zanRa84k5TqdPbTekpwjPHnwGQoT6XpPKgBBlhGyS6s/edit?usp=sharing

fostermh commented:
this does not appear to be the full list of datacite relations types however? I thought the point was to list all the datacite options and then map them to iso

Br-Johnson commented:
Yeah exactly that's right. I was trying to update it before lunch but didn't get a chance

Br-Johnson commented:
Those updated DataCite to ISO mappings look great @fostermh

fostermh commented:
I think this is ready to go now. I have updated the relationship type to use the datacite list of values and mapped them to the iso list for iso XML output. The datacite relatedIdentifier field is now populated on datacite son output.

Br-Johnson commented:
OK cool. I'll have a look!

Br-Johnson commented on src/components/FormComponents/RelatedWorks.jsx:
```diff
@@ -0,0 +1,266 @@
+                  optionLabels={Object.values(associationTypeCode).map(
+                    ({ title }) => title[language]
+                  )}
+                  optionTooltips={Object.values(associationTypeCode).map(
```
I feel like the current Tooltip that describe how the DataCite relation types map to ISO isn't particularly useful for the user. Could you update the tooltip so it explains the DataCite relation type using the definitions here?: https://datacite-metadata-schema.readthedocs.io/en/4.5/appendices/appendix-1/relationType/#issourceof

Br-Johnson commented on src/components/Tabs/RelatedWorksTab.jsx:
```diff
@@ -0,0 +1,77 @@
+
+                Some of the ways a work can be related are:
+                <ul>
+                  <li>Cross Reference - Reference from one record to another.	Use to identify related documents or related records that may be part of the same collection or project</li>
```
I actually think the description of how works can be related should be removed from the top header, and just explained using the Relation Type examples.

fostermh commented:
replaced by https://github.com/cioos-siooc/metadata-entry-form/pull/322
